sbi/inference/base.py

```diff
@@ -152,6 +154,37 @@ def __init__(
             median_observation_distances=[], epochs=[], best_validation_log_probs=[],
         )
 
+    def provide_presimulated(self, theta: Tensor, x: Tensor) -> None:
+        r"""
+        Provide external $\theta$ and $x$ to be used for training later on.
+
+        In this function, $\theta$ and $x$ are merely stored in an attribute. Later on,
+        they will be merged with simulated data, processed (e.g. to get rid of invalid
+        simulations), and eventually put into the banks to be used for training.
+        """
+        self._external_data = (theta, x)
+        self._presimulated_current_round = True
+
+    def _prepend_presimulated(self, theta: Tensor, x: Tensor) -> Tuple[Tensor, Tensor]:
```

Comment:
Maybe you could rename x and theta here because it looks like we are prepending them, while we are actually prepending the external data that was saved in `self._external_data` somewhere else. 

E.g., you could name the args `{theta, x}_simulated` to distinguish them from the `pre`-simulated theta and x.